File: Assignment12/Assignment12.py
# Name: Assignment12.py
# Description: This script answers basic python syntax and operations from Assignment 12.
# All work is completed by Judah Alter.

# 1. Fit the model
import pandas as pd
import statsmodels.formula.api as smf
plants = pd.read_csv('Datasets:UseCase/plants.txt', delim_whitespace=True)
model = smf.ols('Yield ~ Condition', data=plants).fit()
print(model.summary())

# 2. Display the ANOVA table
import statsmodels.api as sm
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)

File: Assignment12/Assignment12.R
# Name: Assignment12.R
# Description: This script answers basic R syntax and operations from Assignment 12.
# All work is completed by Judah Alter.

plants <- read.table("Datasets:UseCase/plants.txt", header=TRUE)
plants$Condition <- as.factor(plants$Condition)  # Convert to factor

# Fit the model
model <- lm(Yield ~ Condition, data=plants)
print(summary(model))

# Display the ANOVA table
print(anova(model))

# Tukey's pairwise comparisons
library(multcomp)
tukey <- glht(model, linfct = mcp(Condition = "Tukey"))
summary(tukey)